kanalony-storage-logic/src/main/scala/QueryParamsValidator.scala

```diff
@@ -0,0 +1,81 @@
+package kanalony.storage.logic
+
+import kanalony.storage.logic.queries.model.DimensionEqualityConstraint
+
+/**
+ * Created by elad.benedict on 2/11/2016.
+ */
+
+class QueryParamterException(message : String) extends IllegalArgumentException(message)
+
+case object QueryParamsValidator {
+
+  private def extractEqualityConstraintParam[T](dimension : Dimensions.Value, queryParams : QueryParams): List[T] = {
+    val dimDef = queryParams.dimensionDefinitions.find(_.dimension == dimension)
+    try {
+      dimDef match {
+        case Some(eqCons) => {
```

Comment:
I can see why it would be preferable to "flatmap that s***" as part of an ongoing computation, however this seems appropriate here, as it is the end of the computation chain. Let's discuss it if you still think otherwise